### PR TITLE
Add test that requires clean style to pass.

### PR DESCRIPTION
**Pull request by destijl**
- Run yapf over all code files
- Don't split before first arg to match Google style

---
This change is [Reviewable](https://reviewable.io/reviews/forensicartifacts/artifacts/188)

### CODE REVIEW

File: .style.yapf [hunk 1]
```diff
@@ -15,7 +15,7 @@ SPACES_BEFORE_COMMENT=2
 SPACE_BETWEEN_ENDING_COMMA_AND_CLOSING_BRACKET=False
 SPLIT_ARGUMENTS_WHEN_COMMA_TERMINATED=False
 SPLIT_BEFORE_BITWISE_OPERATOR=True
-SPLIT_BEFORE_FIRST_ARGUMENT=True
+SPLIT_BEFORE_FIRST_ARGUMENT=False
```
Comment (joachimmetz) [inline .style.yapf]: Why? Turning this off creates harder to read constructions like:
+    artifact_definition = artifact.ArtifactDefinition(name,
 -        name, description=description)
+                                                      description=description)

where the treat this as a regular text (sentence) approach is much friendlier (less eye movement required).

File: artifacts/source_type.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
-"""The source type objects.
+r"""The source type objects.
```
Comment (joachimmetz) [inline artifacts/source_type.py]: Why not keep as is `r""""` is not anywhere else?

File: requirements.txt [hunk 3]
```diff
@@ -1,2 +1,3 @@
 PyYAML==3.11
+yapf==0.10.0
```
Comment (joachimmetz) [inline requirements.txt]: This is not a dependency! Move to setup_requires (or equiv)

https://pip.pypa.io/en/latest/reference/pip_install/#controlling-setup-requires

File: tests/style_test.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+"""Enforce code style."""
+
+import subprocess
+import unittest
+
+from artifacts import errors
+
+
+class StyleTest(unittest.TestCase):
```
Comment (joachimmetz) [inline tests/style_test.py]: + docstring so test information is shown during tests

File: tests/style_test.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  def testCodeStyle(self):
```
Comment (joachimmetz) [inline tests/style_test.py]: + docstring so test information is shown during tests

File: artifacts/artifact.py [hunk 6]
```diff
@@ -75,14 +75,14 @@ def AppendSource(self, type_indicator, attributes):
...
-          u'Unsupported type indicator: {0}.'.format(type_indicator))
+      raise errors.FormatError(u'Unsupported type indicator: {0}.'.format(
+          type_indicator))
 
     try:
       source_object = source_type_class(**attributes)
     except (TypeError, AttributeError) as e:
       raise errors.FormatError(
-          "Invalid artifact definition for {0}: {1}".format(self.name, e))
+          'Invalid artifact definition for {0}: {1}'.format(self.name, e))
```
Comment (joachimmetz) [inline artifacts/artifact.py]: Please add: u'...'

File: .style.yapf (hunk 1 above)
Comment (destijl) [inline .style.yapf]: That's the problem with style, everyone thinks slightly differently, so I'd like to just adopt the google standard as closely as possible. I made this change to make it align better with google style so my editor set to enforce google style doesn't fight with yapf. I've asked the author if he can supply a yapf style file that mimicks internal style, we should adopt that if one exists. If not, this change at least fixed the most obvious difference.

File: artifacts/source_type.py (hunk 2 above)
Comment (destijl) [inline artifacts/source_type.py]: the google linter complains about it.

File: requirements.txt (hunk 3 above)
Comment (destijl) [inline requirements.txt]: I considered this, the problem is that most people who install this code will want to run the tests to validate any artifacts they write. Which will now break if this isn't installed. The way we solve this in grr is to have a separate package with test dependencies, but that is overkill in this case. I think adding a small dependency here is less bad than having tests break unexpectedly. I don't feel too strongly though, I can install it separately in the travis config if you like.

File: tests/style_test.py (hunk 4 above)
Comment (destijl) [inline tests/style_test.py]: done

File: tests/style_test.py (hunk 5 above)
Comment (destijl) [inline tests/style_test.py]: done

File: artifacts/source_type.py (hunk 2 above)
Comment (joachimmetz) [inline artifacts/source_type.py]: So this deviates from pylint

File: requirements.txt (hunk 3 above)
Comment (joachimmetz) [inline requirements.txt]: As discussed thought this applied to setup.py but only applies to travis configuration.

File: .style.yapf (hunk 1 above)
Comment (joachimmetz) [inline .style.yapf]: Google style guide allows for both. We are not following the [Google Python style guide](https://google.github.io/styleguide/pyguide.html) to the letter.

File: artifacts/definitions.py [hunk 7]
```diff
@@ -34,9 +34,8 @@
     'Cloud Storage': 'Cloud storage artifacts.',
     'Configuration Files': 'Configuration files artifacts.',
     'Execution': 'Contain execution events.',
-    'ExternalAccount': (
-        'Information about any users\' account, e.g. username, '
-        'account ID, etc.'),
+    'ExternalAccount': ('Information about any user accounts e.g. username, '
+                        'account ID, etc.'),
     'External Media': 'Contain external media data or events e.g. USB drives.',
     'KnowledgeBase': 'Artifacts used in knowledge base generation.',
```
Comment (joachimmetz) [inline artifacts/definitions.py]: Please remove duplicate key: KnowledgeBase

************* Module artifacts.definitions
W: 29, 9: Duplicate key 'KnowledgeBase' in dictionary (duplicate-key)

**Comment by joachimmetz** [comment]
Per discussion let's set up the editor so that it uses the right versions of the tools.

LGTM

**Pull request merged**